### ISSUE TITLE
Some Klee spline shapes do not match C2C plotter.

### ISSUE THREAD

**Issue by BradWhitlock** [LLNL/axom#979]
I was making a new dataset using C2C format and noticed that for some contours that use sides made of RZ splines, the shapes output by Klee do not match the c2c_plotter program. The Klee version of the curved edge is very wavy compared to the C2C version. Here is such a shape:

Maybe it is a difference in how the splines are sampled parametrically?

black3.contour:
```
piece = line(start=(3cm,143cm), end=(4cm,149cm))
piece = rz(units=cm,
  rz=4 149
     17 151
     33 155
     43 155
     46 153
)

piece = line(end=(44cm,146cm))
piece = line(end=(38cm,147cm))
piece = rz(units=cm,
  rz=38 147
     33 147
     26 146
)

piece = line(end=(21cm,144cm))
piece = line(end=(3cm,143cm))
```

black3.yaml (needed for klee):
```
dimensions: 2

shapes:
- name: black3
  material: black
  geometry:
    format: c2c
    path: black3.contour
```

Plotting file using C2C:
```
/usr/gapps/c2c/latest/bin/c2c_plotter black3.contour
```

shaping in klee (with sampling):
```
data_collection_util --min 140 0 --max 170 50 --res 150 200 -p 1
../examples/quest_shaping_driver_ex --mesh-file box_2d.root --shape-file black3.yaml --method 0 -v -o 0 -p omp

```

**Comment by BradWhitlock** [comment]
Here are some examples of what I mean:
![black3](https://user-images.githubusercontent.com/2072964/207134865-9879b681-f4d2-47f9-9cb5-15579e950b39.png)
![black3_shaped](https://user-images.githubusercontent.com/2072964/207134883-c65f842f-f0f0-4e82-a680-2860b0168d50.png)

**Comment by kennyweiss** [comment]
Thanks for reporting this @BradWhitlock -- Assuming the c2c plotter is correct, it looks like there might be a bug in how we're converting the input spline to NURBS or in how we're sampling the NURBS curve.

**Comment by rhornung67** [comment]
Fixed in new C2C. Wait for next C2C update, which is imminent.

**Comment by kennyweiss** [comment]
I followed the above steps using c2c@1.8.0 and confirmed that the problem is resolved in #1092
Here's a screenshot of the `heroic_roses/black3.contour` using c2c@1.8.0:

![image](https://github.com/LLNL/axom/assets/5626552/b748422b-7eb5-4d12-b0d0-6e61316fe05e)
